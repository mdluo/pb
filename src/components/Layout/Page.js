import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Layout, Card, Icon } from 'antd';

import styles from './Page.module.less';

const { Footer, Content } = Layout;

const Link = ({ href, children }) => (
  <a href={href} target="_blank" rel="noopener noreferrer">{children}</a>
);

class Page extends Component {
  render() {
    const { children, loading } = this.props;
    return (
      <Layout className={styles.layout}>
        <Content>
          <Card
            className={styles.card}
            bordered={false}
            title="Progress Bar Demo"
            extra={<Link href="https://github.com/mdluo/pb"><Icon type="link" /> GitHub</Link>}
            loading={loading}
          >
            {children}
          </Card>
        </Content>
        <Footer>
          <p>Powered by <Link href="https://github.com/facebook/create-react-app">CRA</Link>, <Link href="https://ant.design/">Antd</Link> and more.</p>
          <p>Made with <Icon className={styles.heart} type="heart" /> by <Link href="https://github.com/mdluo">mdluo</Link>.</p>
        </Footer>
      </Layout>
    );
  }
}

Page.propTypes = {
  children: PropTypes.node.isRequired,
};

Page.defaultProps = {
  loading: false,
};

export default Page;
