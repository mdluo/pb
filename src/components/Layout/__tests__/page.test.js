import React from 'react';
import { render } from 'enzyme';

import Page from '../Page';

describe('Bar', () => {
  it('renders correctly', () => {
    const wrapper = render(<Page>Content</Page>);
    expect(wrapper).toMatchSnapshot();
  });
});
