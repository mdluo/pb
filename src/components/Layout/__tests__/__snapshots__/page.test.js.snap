// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar renders correctly 1`] = `
<div
  class="layout ant-layout"
>
  <div
    class="ant-layout-content"
  >
    <div
      class="ant-card card"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Progress Bar Demo
          </div>
          <div
            class="ant-card-extra"
          >
            <a
              href="https://github.com/mdluo/pb"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="anticon anticon-link"
              />
               GitHub
            </a>
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        Content
      </div>
    </div>
  </div>
  <div
    class="ant-layout-footer"
  >
    <p>
      Powered by 
      <a
        href="https://github.com/facebook/create-react-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        CRA
      </a>
      , 
      <a
        href="https://ant.design/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Antd
      </a>
       and more.
    </p>
    <p>
      Made with 
      <i
        class="anticon anticon-heart heart"
      />
       by 
      <a
        href="https://github.com/mdluo"
        rel="noopener noreferrer"
        target="_blank"
      >
        mdluo
      </a>
      .
    </p>
  </div>
</div>
`;
